### PR TITLE
FSDK-103: Typescript StatusUpdateForm

### PR DESCRIPTION
**Pull request by mahboubii**

### CODE REVIEW

File: src/components/StatusUpdateForm/StatusUpdateForm.tsx [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    <Panel>
+      <form onSubmit={state.onSubmitForm}>
+        <ImageDropzone handleFiles={state.uploadNewFiles}>
+          <PanelHeading>{Header ?? <Title>{t('New Post')}</Title>}</PanelHeading>
+
+          <PanelContent>
+            <div style={{ display: 'flex' }}>
+              {state.userData.profileImage && (
+                <div style={{ marginRight: '16px' }}>
+                  <Avatar image={state.userData.profileImage || 'https://placehold.it/100x100'} size={50} circle />
```
Comment (arnautov-anton) [inline src/components/StatusUpdateForm/StatusUpdateForm.tsx:99]:
```suggestion
                  <Avatar image={state.userData.profileImage} size={50} circle />
```

File: src/components/StatusUpdateForm/useStatusUpdateForm.ts [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+
+  useEffect(() => {
+    files.order
+      .filter((id) => !reqInProgress.current[id] && files.data[id].state === 'uploading')
+      .forEach(async (id) => {
+        reqInProgress.current[id] = true;
+        await uploadFile(id, files.data[id]);
+        delete reqInProgress.current[id];
+      });
+  }, [files.order]);
```
Comment (arnautov-anton) [inline src/components/StatusUpdateForm/useStatusUpdateForm.ts:347]: These two look identical, might be a good idea to create one parametrised function for it.

File: src/components/StatusUpdateForm/useStatusUpdateForm.ts [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+      if (onSuccess) onSuccess(response);
+    } catch (e) {
+      setSubmitting(false);
+      logErr(e, 'add-activity');
+    }
+  };
+
+  const onChange = useCallback((event: SyntheticEvent<HTMLTextAreaElement>) => {
+    const text = inputValueFromEvent(event);
+    if (text == null) return;
```
Comment (arnautov-anton) [inline src/components/StatusUpdateForm/useStatusUpdateForm.ts:486]:
```suggestion
    if (text === null) return;
```

File: src/components/StatusUpdateForm/useStatusUpdateForm.ts (hunk 2 above)
Comment (mahboubii) [reply src/components/StatusUpdateForm/useStatusUpdateForm.ts:347]: at this point refactoring these two parts actually makes the code harder to follow IMO

Comment (mahboubii) [reply src/components/StatusUpdateForm/useStatusUpdateForm.ts:347]: btw, please don't force push to a branch which is not local anymore

**Pull request merged**